### PR TITLE
Include DM particles information in NTuples.

### PR DESCRIPTION
**Pull request by alopezso**

### CODE REVIEW

File: data/DMHiggsAnalysis.cfg [hunk 1]
```diff
@@ -13,13 +13,12 @@ SampleName:       sample
 InputFileList:    input_data.txt
 BaseConfig:       HGamAnalysisFramework/HGamRel20p7.config
 Include:          HGamAnalysisFramework/MCSamples.config
-
+NTupleName:	  ggH125_gamgam.NTuple.root
```
Comment (rjwang) [inline data/DMHiggsAnalysis.cfg:16]: what is this for ?

File: Root/DMHiggsAnalysis.cxx [hunk 2]
```diff
@@ -31,457 +31,587 @@ ClassImp(DMHiggsAnalysis)
...
+bool DMHiggsAnalysis::isTruthPhoton(const xAOD::TruthParticle* truth){
 
+  if( !truth ) return false;
+  if( fabs(truth->pdgId()) != 22 ) return false;
+
+  return true;    
+    
+}
+
+bool DMHiggsAnalysis::isTruthHiggsPhoton(const xAOD::TruthParticle* truth){
```
Comment (rjwang) [inline Root/DMHiggsAnalysis.cxx:61]: this function has nothing inside, do we need to add?

File: Root/DMHiggsAnalysis.cxx [hunk 3]
```diff
@@ -31,457 +31,587 @@ ClassImp(DMHiggsAnalysis)
...
-    myEvents->Branch("electron_Pz", electron_Pz,"electron_Pz[nElectrons]/F");
-    myEvents->Branch("electron_E",  electron_E, "electron_E[nElectrons]/F");
-    myEvents->Branch("electron_charge",electron_charge,"electron_charge[nElectrons]/I");
+  std::string inputfileName = wk()->inputFileName();
+  std::string numberFile = inputfileName;
+  //currentfilename = inputfileName;
+  if( config()->getBool("IsMxAOD") )
+    m_isMxAOD=true;
+  else
+    m_isMxAOD=false;
```
Comment (rjwang) [inline Root/DMHiggsAnalysis.cxx:201]: I don't think we need this, when adding the gen-level information, we are always running on mXAOD

**Review by rjwang — CHANGES_REQUESTED** [review]
there are several changes which are not relevant to the addition of generator level, please go to clean them.

File: data/DMHiggsAnalysis.cfg (hunk 1 above)
Comment (alopezso) [reply data/DMHiggsAnalysis.cfg:16]: The problem is that DM particles are not added on MxAOD. So, everything needs to be done on DxAODs. But f you run on MxAODs, the inputfile name won't be sample name but something of style DAOD_HIGGS1..... So we need to include in data/DMHiggsAnalysis.cfg the output NTuple name and not derive it from the inputName

File: Root/DMHiggsAnalysis.cxx (hunk 2 above)
Comment (alopezso) [reply Root/DMHiggsAnalysis.cxx:61]: No, I added it from a previous code but forgot to eliminate it

File: Root/DMHiggsAnalysis.cxx (hunk 3 above)
Comment (alopezso) [reply Root/DMHiggsAnalysis.cxx:201]: Yes, we need it because of what I said in previous comment

File: Root/DMHiggsAnalysis.cxx (hunk 2 above)
Comment (alopezso) [reply Root/DMHiggsAnalysis.cxx:61]: I have just eliminate it and did another commit. It is inside the pull request

File: data/DMHiggsAnalysis.cfg (hunk 1 above)
Comment (rjwang) [reply data/DMHiggsAnalysis.cfg:16]: okay, if I only run MxAODs, is there a way to get the gen_MET ?

Comment (alopezso) [reply data/DMHiggsAnalysis.cfg:16]: Yes, getting truth MET either accessing through the truthHandler or directly on the containers or HGamVariables. This met was estimated with all non-interacting particles, not just DM particles.

Comment (rjwang) [reply data/DMHiggsAnalysis.cfg:16]: okay, then I understand the situation, just to double check, if you run on MxAOD, get the genMET from  all non-interacting particles. And the same signal sample, but DxAODs, get the genMET from DM particles, are the two plots same ?

File: Root/DMHiggsAnalysis.cxx (hunk 3 above)
Comment (rjwang) [reply Root/DMHiggsAnalysis.cxx:201]: okay, i see

**Comment by rjwang** [comment]
what is the status of adding the genMET now?

**Comment by alopezso** [comment]
Hi ! I have just done this commit : https://github.com/rjwang/DMHiggsAnalysis/pull/31/commits/544fedc5e065bc6df527298f113f43fa7436a141

**Comment by rjwang** [comment]
this commit adds both ways retrieve the gen-level information?  One is from DAOD, one is from MxAOD, they are not working at the same time, right ? If this is true, please keep only the way from MxAOD, totally remove the one from DAOD.

**Comment by alopezso** [comment]
Hi Renjie,
Sorry for answering so late but I was doing the MxAODSys and couldn't test the DMHiggsAnalysis. However, I think that if you want that, you could just reject the pull request, since it only uses MxAOD.

**Comment by rjwang** [comment]
I am closing this pull request, please go to add a new one ONLY with gen MET running on MxAOD

**Pull request closed without merging**